### PR TITLE
Added builder info

### PR DESCRIPTION
**Pull request by kwojcicki**
Signed-off-by: Krystian Wojcicki <kwojcicki@sympatico.ca>

Client part of https://github.com/moby/moby/issues/38518

**- What I did**
Added the builder version in the docker info command

**- How I did it**
reading the response from the server and printing out the corresponding version
also updated the version of moby/moby being used to get the new struct information.

**- How to verify it**
typing docker info/the tests

**- Description for the changelog**
Docker info prints the default builder 

**- A picture of a cute animal (not mandatory but encouraged)**
image

### CODE REVIEW

**Comment by silvin-lubecki** [comment]
Hello @kwojcicki , thank you for this PR!
I can see the CI is failing on the `validate` step:
```
2019/02/05 04:56:25 WARNING: package github.com/Nvveen/Gotty is unused, consider removing it from vendor.conf
```
and on `test` step:
```
=== Failed
=== FAIL: cli/command/system TestPrettyPrintInfo/info_without_swarm (0.00s)
    --- FAIL: TestPrettyPrintInfo/info_without_swarm (0.00s)
        info_test.go:340: assertion failed: 
            --- expected
            +++ actual
            @@ -8,4 +8,5 @@
               Stopped: 0
              Images: 0
            + Default Builder: Legacy
              Server Version: 17.06.1-ce
              Storage Driver: aufs
            

=== FAIL: cli/command/system TestPrettyPrintInfo/info_with_plugins (0.00s)
    --- FAIL: TestPrettyPrintInfo/info_with_plugins (0.00s)
        info_test.go:340: assertion failed: 
            --- expected
            +++ actual
            @@ -10,4 +10,5 @@
               Stopped: 0
              Images: 0
            + Default Builder: Legacy
              Server Version: 17.06.1-ce
              Storage Driver: aufs
            

=== FAIL: cli/command/system TestPrettyPrintInfo/info_with_swarm (0.00s)
    --- FAIL: TestPrettyPrintInfo/info_with_swarm (0.00s)
        info_test.go:340: assertion failed: 
            --- expected
            +++ actual
            @@ -8,4 +8,5 @@
               Stopped: 0
              Images: 0
            + Default Builder: Legacy
              Server Version: 17.06.1-ce
              Storage Driver: aufs
            

=== FAIL: cli/command/system TestPrettyPrintInfo/info_with_legacy_warnings (0.00s)
    --- FAIL: TestPrettyPrintInfo/info_with_legacy_warnings (0.00s)
        info_test.go:340: assertion failed: 
            --- expected
            +++ actual
            @@ -8,4 +8,5 @@
               Stopped: 0
              Images: 0
            + Default Builder: Legacy
              Server Version: 17.06.1-ce
              Storage Driver: aufs
            

=== FAIL: cli/command/system TestPrettyPrintInfo/info_with_daemon_warnings (0.00s)
    --- FAIL: TestPrettyPrintInfo/info_with_daemon_warnings (0.00s)
        info_test.go:340: assertion failed: 
            --- expected
            +++ actual
            @@ -8,4 +8,5 @@
               Stopped: 0
              Images: 0
            + Default Builder: Legacy
              Server Version: 17.06.1-ce
              Storage Driver: aufs
            

=== FAIL: cli/command/system TestPrettyPrintInfo/bad_security_info (0.00s)
    --- FAIL: TestPrettyPrintInfo/bad_security_info (0.00s)
        info_test.go:340: assertion failed: 
            --- expected
            +++ actual
            @@ -8,4 +8,5 @@
               Stopped: 0
              Images: 0
            + Default Builder: Legacy
              Server Version: 17.06.1-ce
              Storage Driver: aufs
```

File: cli/command/system/info.go [hunk 1]
```diff
@@ -143,6 +143,11 @@ func prettyPrintServerInfo(dockerCli command.Cli, info types.Info) []error {
 	fmt.Fprintln(dockerCli.Out(), "  Paused:", info.ContainersPaused)
 	fmt.Fprintln(dockerCli.Out(), "  Stopped:", info.ContainersStopped)
 	fmt.Fprintln(dockerCli.Out(), " Images:", info.Images)
+	if info.Builder == types.BuilderV1 || info.Builder == "" {
```
Comment (silvin-lubecki) [inline cli/command/system/info.go]: What do you think about this?
```go
builder := "Unknown"
switch info.Builder{
case "", types.BuilderV1:
    builder = "Legacy"
case types.BuilderBuildKit:
    builder = "Buildkit"
default:
    errs = append(errs, fmt.Errorf("unknown builder type %q", info.Builder)
}
fmt.Fprintln(dockerCli.Out(), " Default Builder:", builder)
```

Comment (kwojcicki) [reply cli/command/system/info.go]: I agree that looks much cleaner, switched to that 😄

**Comment by kwojcicki** [comment]
Hey @silvin-lubecki think I correctly updated the .golden files and the vendor.conf file

**Comment by silvin-lubecki** [comment]
Closed in favor of #1660

**Pull request closed without merging**